src/spi.rs

```diff
@@ -136,10 +237,25 @@ impl spi::Write<u8> for Mock {
     }
 }
 
-impl FullDuplex<u8> for Mock {
-    type Error = ErrorKind;
+impl spi::SpiBusRead<u8> for Mock {
+    /// spi::Read implementation for Mock
+    ///
+    /// This will cause an assertion if the read call does not match the next expectation
+    fn read(&mut self, buffer: &mut [u8]) -> Result<(), Self::Error> {
+        let w = self.next().expect("no expectation for spi::read call");
+        assert_eq!(w.expected_mode, Mode::Read, "spi::read unexpected mode");
+        assert_eq!(
+            buffer.len(),
+            w.response.len(),
+            "i2c:read mismatched response length"
```

Comment:
```suggestion
            "spi:read mismatched response length"
```

Not a maintainer, just found this bug while using this branch.

Thanks for updating this!